docs/src/callbacks.md

```diff
@@ -0,0 +1,102 @@
+* The [`SteadyStateCallback`](@ref) terminates the time integration when the residual steady state
+  falls below a certain threshold. This checks the convergence of the potential ``\phi`` for
+  hyperbolic diffusion. See e.g. [examples/2d/elixir\_hypdiff\_nonperiodic.jl](https://github.com/trixi-framework/Trixi.jl/blob/main/examples/2d/elixir_hypdiff_nonperiodic.jl).
+* The [`GlmSpeedCallback`](@ref) updates the divergence cleaning wave speed `c_h` for the ideal
+  GLM-MHD equations. See e.g. [examples/2d/elixir\_mhd\_alfven\_wave.jl](https://github.com/trixi-framework/Trixi.jl/blob/main/examples/2d/elixir_mhd_alfven_wave.jl).
+
+## Usage of Step Callbacks
+Step callbacks are passed to the `solve` method from the ODE solver via the keyword argument
+`callback`. If you want to use a single callback `cb`, pass it as `callback=cb`. When using two or
+more callbacks, you need to turn them into a `CallbackSet` first by calling
+`callbacks = CallbackSet(cb1, cb2)` and passing it as `callback=callbacks`.
+
+!!! note
+    There are some restrictions regarding the order of callbacks in a `CallbackSet`.
+
+    The callbacks are called *after* each time step but some callbacks actually belong to the next
+    time step. Therefore, the callbacks should be ordered in the following way:
+    * Callbacks that belong to the current time step:
+        * `SummaryCallback` controls, among other things, timers and should thus be first
+        * `SteadyStateCallback` may mark a time step as the last one
+        * `AnalysisCallback` may do some checks that mark a time step as the last one
+        * `AliveCallback` should be nearby `AnalysisCallback`
+        * `SaveSolutionCallback`/`SaveRestartCallback` should save the current solution before it is
+          degraded by AMR
+        * `VisualizationCallback` should be called before the mesh is adapted
+    * Callbacks that belong to the next time step:
+        * `AMRCallback`
+        * `StepsizeCallback` must be called after `AMRCallback` to accomodate potential changes to
+          the mesh
+        * `GlmSpeedCallback` must be called after `StepsizeCallback` because the step size affects
+          the value of `c_h`
+        * `LBMCollisionCallback` is already part of the calculations of the next time step and
+          should therefore be called after `StepsizeCallback`
+
+## Stage Callbacks
+[`PositivityPreservingLimiterZhangShu`](@ref) is a positivity-preserving limiter, used to enforce
+physical constraints. An example elixir using this feature can be found at
+[examples/2d/elixir\_euler\_positivity.jl](https://github.com/trixi-framework/Trixi.jl/blob/main/examples/2d/elixir_euler_positivity.jl).
+
+## Implementing New Callbacks
+Since Trixi is compatible with [OrdinaryDiffEq.jl](https://github.com/SciML/OrdinaryDiffEq.jl),
+both packages share the same callback interface. A detailed description of it can be found in the
+OrdinaryDiffEq.jl documentation.
```

Comment:
Can you link to this detailed description in their docs please?

Reply:
Done in fe783972770345b30689982be2f4a293eeffd6c9.